.ci/magician/github/reviewer_assignment.go

```diff
@@ -66,3 +67,24 @@ func FormatReviewerComment(newPrimaryReviewer string) string {
 	})
 	return sb.String()
 }
+
+var reviewerCommentRegex = regexp.MustCompile("@(?P<reviewer>[^,]*), a repository maintainer, has been assigned to review your changes.")
+
+// FindReviewerComment returns the comment which mentions the current primary reviewer and the reviewer's login,
+// or an empty comment and empty string if no such comment is found.
+// comments should only include comments by the magician in the current PR.
+func FindReviewerComment(comments []PullRequestComment) (PullRequestComment, string) {
+	for _, comment := range comments {
```

Comment:
what order are the comments returned in? first -> last?

If so, should it be reversed so that the latest reviewer comment is used, and not the first

Reply:
Not sure if it's returned in any particular order. Probably the best thing would be to compare the creation times.

Reply:
Just minor detail, small chance the user may have to run the command twice if there's been multiple reviewers

Reply:
@c2thorn there should only ever be one comment that contains the reviewer - the proposed implementation is to update that comment when the reviewer is reassigned (so that we're not creating more noise in the PR). I don't _think_ there's a way for us to end up with more than one, but FWIW if it ends up being a common issue I'd prefer to delete or hide the duplicates.

Reply:
Just in case, I've made it return the most recent reviewer comment.

Reply:
IMO It should take the oldest reviewer comment rather than the most recent. But since there should only be one it's not a blocker for me at this time.